test/test_irods_http_api.py

```diff
@@ -746,6 +746,198 @@ def test_common_operations(self):
+                'op': 'stat',
+                'lpath': data_object_b
+            })
+            logging.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertGreater(result['size'], 0)
+
+            # Capture the size of "data_object_b".
+            # We'll need this to prove option, "update_existing", works.
+            size_of_data_object_b = result['size']
+
+            # Show attempting to copy over an existing data object using "update_existing" as the
+            # value to the "option" parameter results in different outcomes depending on the mtimes
+            # of the data objects.
+            logging.debug('option=update_existing (a over b)')
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'copy',
+                'src-lpath': data_object_a,
+                'dst-lpath': data_object_b,
+                'option': 'update_existing'
+            })
+            logging.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertFalse(result['copied'])
+
+            # "data_object_b" is still not empty.
+            r = requests.get(self.url_endpoint, headers=rodsuser_headers, params={
+                'op': 'stat',
+                'lpath': data_object_b
+            })
+            logging.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertGreater(result['size'], 0)
+
+            logging.debug('option=update_existing (b over a)')
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'copy',
+                'src-lpath': data_object_b,
+                'dst-lpath': data_object_a,
+                'option': 'update_existing'
+            })
+            logging.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertTrue(result['copied'])
+
+            # "data_object_a" is no longer empty.
+            r = requests.get(self.url_endpoint, headers=rodsuser_headers, params={
+                'op': 'stat',
+                'lpath': data_object_a
+            })
+            logging.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertEqual(result['size'], size_of_data_object_b)
+
+            # Write some new data to "data_object_a".
+            # This is how we prove the "option" value, "overwrite_existing" works.
```

Comment:
Is this supposed to say "update_existing" instead of "overwrite_existing", or is this just saying that these two options work the same way in this particular sense?

Reply:
The case for `update_existing` was covered before this point.

This is talking about preparing the ground for the `overwrite_existing` case.

Does that answer the question?